cardano-node/src/Cardano/Node/Tracing/Tracers/P2P.hs

```diff
@@ -0,0 +1,1375 @@
+             , "domainAddress" .= toJSON d
+             , "result" .= toJSONList res
+             ]
+  forMachine _dtal (TraceLocalRootGroups groups) =
+    mkObject [ "kind" .= String "LocalRootGroups"
+             , "localRootGroups" .= toJSON groups
+             ]
+  forMachine _dtal (TraceLocalRootFailure d exception) =
+    mkObject [ "kind" .= String "LocalRootFailure"
+             , "domainAddress" .= toJSON d
+             , "reason" .= show exception
+             ]
+  forMachine _dtal (TraceLocalRootError d exception) =
+    mkObject [ "kind" .= String "LocalRootError"
+             , "domainAddress" .= toJSON d
+             , "reason" .= show exception
+             ]
+  forHuman = pack . show
+
+docLocalRootPeers :: Documented (TraceLocalRootPeers ntnAddr resolverError)
+docLocalRootPeers = Documented [
+    DocMsg
+      (TraceLocalRootDomains anyProto)
+      []
+      ""
+  , DocMsg
+      (TraceLocalRootWaiting anyProto anyProto)
+      []
+      ""
+  , DocMsg
+      (TraceLocalRootResult anyProto anyProto)
+      []
+      ""
+  , DocMsg
+      (TraceLocalRootGroups anyProto)
+      []
+      ""
+  , DocMsg
+      (TraceLocalRootFailure anyProto anyProto)
+      []
+      ""
+  , DocMsg
+      (TraceLocalRootError anyProto anyProto)
+      []
+      ""
+  ]
+
+--------------------------------------------------------------------------------
+-- PublicRootPeers Tracer
+--------------------------------------------------------------------------------
+
+namesForPublicRootPeers :: TracePublicRootPeers -> [Text]
+namesForPublicRootPeers TracePublicRootRelayAccessPoint {} = ["PublicRootRelayAccessPoint"]
+namesForPublicRootPeers TracePublicRootDomains {} = ["PublicRootDomains"]
+namesForPublicRootPeers TracePublicRootResult {}  = ["PublicRootResult"]
+namesForPublicRootPeers TracePublicRootFailure {}  = ["PublicRootFailure"]
+
+severityPublicRootPeers :: TracePublicRootPeers -> SeverityS
+severityPublicRootPeers _ = Info
+
+instance LogFormatting TracePublicRootPeers where
+  forMachine _dtal (TracePublicRootRelayAccessPoint relays) =
+    mkObject [ "kind" .= String "PublicRootRelayAddresses"
+             , "relayAddresses" .= toJSONList relays
+             ]
+  forMachine _dtal (TracePublicRootDomains domains) =
+    mkObject [ "kind" .= String "PublicRootDomains"
+             , "domainAddresses" .= toJSONList domains
+             ]
+  forMachine _dtal (TracePublicRootResult b res) =
+    mkObject [ "kind" .= String "PublicRootResult"
+             , "domain" .= show b
+             , "result" .= toJSONList res
+             ]
+  forMachine _dtal (TracePublicRootFailure b d) =
+    mkObject [ "kind" .= String "PublicRootFailure"
+             , "domain" .= show b
+             , "reason" .= show d
+             ]
+  forHuman = pack . show
+
+docPublicRootPeers :: Documented TracePublicRootPeers
+docPublicRootPeers = Documented [
+    DocMsg
+      (TracePublicRootRelayAccessPoint anyProto)
```

Comment:
Was this intentionally left empty?

Reply:
Yes it was intentionally, we can not provide documentation for all tracers.